pkg/doc/did/doc.go

```diff
@@ -118,6 +120,31 @@ type PublicKey struct {
 	jsonWebKey *jose.JWK
 }
 
+// NewPublicKeyFromBytes creates a new PublicKey based on raw public key bytes.
+func NewPublicKeyFromBytes(id, kType, controller string, value []byte) *PublicKey {
+	return &PublicKey{
+		ID:         id,
+		Type:       kType,
+		Controller: controller,
+		Value:      value,
+	}
+}
+
+// NewPublicKeyFromJWK creates a new PublicKey based on JSON Web Key.
+func NewPublicKeyFromJWK(id, kType, controller string, jwk *jose.JWK) *PublicKey {
+	return &PublicKey{
+		ID:         id,
+		Type:       kType,
+		Controller: controller,
+		jsonWebKey: jwk,
```

Comment:
Do we not expose a value based on the JWK?
@kdimak @rolsonquadras @sudeshrshetty

Reply:
It's exposed through a public key method - https://github.com/hyperledger/aries-framework-go/pull/1597/files#diff-ca45993820e1e4b4338acd4e485ae699R144

Reply:
But now there is an exposed Value field that is nil for this case?

Reply:
Yes.

Reply:
From the code, it looks like Value is always set and jsonWebKey is available through publicKey method for some cases - https://github.com/hyperledger/aries-framework-go/blob/47364c99607be0ebeea2395d9e7e97be0f1ed94d/pkg/doc/did/doc.go#L430-L436

Reply:
But if `did.Doc` is created manually using `NewPublicKeyFromJWK()` then `Value` is `nil`. To be consistent, we can convert `JWK` to bytes and fill in the `Value`.

Reply:
Updated `NewPublicKeyFromJWK()`. So now `PublicKey.Value` is always not nil. @rolsonquadras @troyronda